### PR TITLE
[RO-315] Added groupId to line item

### PR DESCRIPTION
**Pull request by luisfn**
The goal of this PR is to introduce a group identifier to line_items table and allow us to execute operations affecting multiple rows instead of single ones.

Ticket: https://oat-sa.atlassian.net/browse/RO-315

### Added
- Added `groupId` column to `line_items` database table.

### Changed
- Updated commands to work with line item group id.
- Renamed `roster:modify-entity:line-item:change-state` to `roster:modify-entity:line-item:change-status`.

### CODE REVIEW

**Review by uncleempty — CHANGES_REQUESTED** [review]
Good job! :+1: 
Just one minor comment + quite big topic:
wdyt, maybe we can just rename `groupId` field to smth more meaningful?

File: src/Command/ModifyEntity/LineItem/LineItemChangeDatesCommand.php [hunk 1]
```diff
@@ -156,22 +171,37 @@ protected function initialize(InputInterface $input, OutputInterface $output): v
...
                     self::OPTION_LINE_ITEM_IDS,
-                    self::OPTION_LINE_ITEM_SLUGS
+                    self::OPTION_LINE_ITEM_SLUGS,
+                    self::OPTION_LINE_ITEM_GROUP_IDS,
                 )
             );
         }
 
-        if (!empty($this->lineItemIds) && !empty($this->lineItemSlugs)) {
+        if ($usedOptions > 1) {
```
[the remark this answers is not in the document]
Comment (luisfn) [reply src/Command/ModifyEntity/LineItem/LineItemChangeDatesCommand.php:198]: We should have only one of then informed. If none or more than one are informed, exceptions are triggered accordingly

https://github.com/oat-sa/simple-roster/blob/feature/RO-315-aggregated-line-items/src/Command/ModifyEntity/LineItem/LineItemChangeDatesCommand.php#L177-L207

File: config/doctrine/LineItem.orm.xml [hunk 2]
```diff
@@ -36,6 +36,7 @@
                 <option name="default">0</option>
             </options>
         </field>
+        <field name="groupId" column="group_id" length="255" nullable="true"/>
```
[the remark this answers is not in the document]
Comment (luisfn) [reply config/doctrine/LineItem.orm.xml:39]: As we discussed via Slack, we will keep it named group_id.

[the remark this answers is not in the document]
Comment (Skyld) [reply config/doctrine/LineItem.orm.xml:39]: Yes I think it's better to keep it as group id, to not increase the cognitive complexity of the application. I think it makes sense to just support logical grouping of users and line items, regardless of what a group means. I think if we will introduce `activity` as a business entity we can replace group id with it, but for now this is enough.

File: src/Command/Ingester/LineItemIngesterCommand.php [hunk 3]
```diff
@@ -110,7 +110,8 @@ protected function execute(InputInterface $input, OutputInterface $output): int
                     'slug',
                     'startTimestamp',
                     'endTimestamp',
-                    'maxAttempts'
+                    'maxAttempts',
+                    'groupId'
```
Comment (Skyld) [inline src/Command/Ingester/LineItemIngesterCommand.php:114]: You can remove it from validation I think, it's optional to specify it in the csv. What do you think?

**Review by Skyld — CHANGES_REQUESTED** [review]

File: tests/Functional/Command/ModifyEntity/LineItem/LineItemChangeDatesCommandTest.php [hunk 4]
```diff
@@ -215,7 +214,8 @@ public function provideInvalidParameters(): array
                     '--start-date' => '2020-01-01T00:00:00+0000',
                     '--end-date' => '2020-01-10T00:00:00+0000',
                 ],
-                'expectedOutput' => 'You need to specify line-item-ids or line-item-slugs option.',
+                'expectedOutput' => 'You need to specify \'line-item-ids\', \'line-item-slugs\'' .
+                    ' or \'line-item-group-ids\' option.',
```
Comment (Skyld) [inline tests/Functional/Command/ModifyEntity/LineItem/LineItemChangeDatesCommandTest.php:218]: Can you please use double quotes in the outside, so you can spare the escapings

**Review by Skyld — CHANGES_REQUESTED** [review]

File: src/Command/Ingester/LineItemIngesterCommand.php (hunk 3 above)
Comment (luisfn) [reply src/Command/Ingester/LineItemIngesterCommand.php]: I agree! Removed

File: tests/Functional/Command/ModifyEntity/LineItem/LineItemChangeDatesCommandTest.php (hunk 4 above)
Comment (luisfn) [reply tests/Functional/Command/ModifyEntity/LineItem/LineItemChangeDatesCommandTest.php]: Fixed! Thanks!

**Review by Skyld — APPROVED** [review]
- [x] New code is covered by tests
- [x] Tests are running successfully (old and new ones) on my local machine
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues
- [x] Feature is working correctly on my local machine
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Changelog is updated according to changes
- [x] Documentation is updated according to changes

**Review by uncleempty — APPROVED** [review]
- [x] New code is covered by tests
- [x] Tests are running successfully (old and new ones) on my local machine
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Changelog is updated according to changes
- [x] Documentation is updated according to changes

**Pull request merged**